nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelRecordReader.java

```diff
@@ -0,0 +1,221 @@
+        }
+
+        return currentRowValues;
+    }
+
+    public static Object getCellValue(Cell cell) {
+        if (cell != null) {
+            switch (cell.getCellType()) {
+                case _NONE:
+                case BLANK:
+                case ERROR:
+                case FORMULA:
+                case STRING:
+                    return cell.getStringCellValue();
+                case NUMERIC:
+                    return DateUtil.isCellDateFormatted(cell) ? cell.getDateCellValue() : cell.getNumericCellValue();
+                case BOOLEAN:
+                    return cell.getBooleanCellValue();
+            }
+        }
+        return null;
+    }
+
+    private Object convert(final Object value, final DataType dataType, final String fieldName) {
+        if (value == null || dataType == null) {
+            return value;
+        }
+
+        return DataTypeUtils.convertType(value, dataType, LAZY_DATE_FORMAT, LAZY_TIME_FORMAT, LAZY_TIMESTAMP_FORMAT, fieldName);
+    }
+
+    private Object convertSimpleIfPossible(final Object value, final DataType dataType, final String fieldName) {
+        if (value == null || dataType == null) {
+            return value;
+        }
+
+        switch (dataType.getFieldType()) {
+            case STRING:
+                return value;
+            case BOOLEAN:
+            case INT:
+            case LONG:
+            case FLOAT:
+            case DOUBLE:
+            case DECIMAL:
+            case BYTE:
+            case CHAR:
+            case SHORT:
+                if (DataTypeUtils.isCompatibleDataType(value, dataType)) {
+                    return DataTypeUtils.convertType(value, dataType, LAZY_DATE_FORMAT, LAZY_TIME_FORMAT, LAZY_TIMESTAMP_FORMAT, fieldName);
+                }
+                break;
+            case DATE:
+                if (DataTypeUtils.isDateTypeCompatible(value, dateFormat)) {
+                    return DataTypeUtils.convertType(value, dataType, LAZY_DATE_FORMAT, LAZY_TIME_FORMAT, LAZY_TIMESTAMP_FORMAT, fieldName);
+                }
+                break;
+            case TIME:
+                if (DataTypeUtils.isTimeTypeCompatible(value, timeFormat)) {
+                    return DataTypeUtils.convertType(value, dataType, LAZY_DATE_FORMAT, LAZY_TIME_FORMAT, LAZY_TIMESTAMP_FORMAT, fieldName);
+                }
+                break;
+            case TIMESTAMP:
+                if (DataTypeUtils.isTimestampTypeCompatible(value, timestampFormat)) {
+                    return DataTypeUtils.convertType(value, dataType, LAZY_DATE_FORMAT, LAZY_TIME_FORMAT, LAZY_TIMESTAMP_FORMAT, fieldName);
+                }
+                break;
+        }
+
+        return value;
+    }
+
+    @Override
+    public RecordSchema getSchema() throws MalformedRecordException {
+        return schema;
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.rowIterator.close();
+    }
+
+    void setLogger(ComponentLog logger) {
```

Comment:
Is there a reason for this setter method as opposed to passing the logger in the constructor?

Reply:
I have this specifically for unit testing so I could instantiate a MockComponentLog using the instantiated `ExcelRecordReader` See line 106 in TestExcelRecordReader

Reply:
Thanks, that is understandable, but optional logging or checking for log statements is generally not the best approach. However, it should be possible to use the NiFi TestRunner to configure and enable the Controller Service, which would provide a ComponentLog that could be checked without wiring the log using this approach.

Reply:
@exceptionfactory What do you mean by

> the Controller Service ?
ExcelRecordReader is not a a controller service

Reply:
I meant to say, the `ExcelReader` Controller Service can be configured and enabled through the TestRunner.

Reply:
Sorry I do not see how. In order to instantiate a TestRunner don't you need a class which implements 
`org.apache.nifi.processor.Processor` ? `ExcelReader` does not implement it.

Reply:
You can use the `NoOpProcessor.class`. See [StandardPrivateKeyServiceTest](https://github.com/apache/nifi/blob/607b53504b57d5d7fde84a721ce75056c92a1f70/nifi-nar-bundles/nifi-standard-services/nifi-key-service-bundle/nifi-key-service/src/test/java/org/apache/nifi/key/service/StandardPrivateKeyServiceTest.java) as an example.